code/game/objects/items/weapons/grenades/chem_grenade.dm

```diff
@@ -17,6 +17,7 @@
 	var/obj/item/slime_extract/C = null	//for Ex grenades
 	var/obj/item/weapon/reagent_containers/glass/beaker/noreactgrenade/reservoir = null
 	var/extract_uses = 0
+	var/mob/primed_by //For logging purposes mostly
```

Comment:
I'd rather you stored the mob name and ckey directly, this can lead to a few hanging references.